internal/pool/pool.go

```diff
@@ -172,16 +197,61 @@ func New[PT Item[T], T any](
 
 	p.createItem = createItemWithTimeoutHandling(p.createItem, p)
 
-	p.idle = make([]PT, 0, p.limit)
-	p.index = make(map[PT]struct{}, p.limit)
 	p.stats = &safeStats{
 		v:        stats.Stats{Limit: p.limit},
 		onChange: p.trace.OnChange,
 	}
 
+	for i := 0; i < defaultSpawnGoroutinesNumber; i++ {
+		go p.spawnItems(ctx)
```

Comment:
context must be avalaible only on New call
After return managment to upper code - context maybe is done
For spawn goroutines - you can use xcontext.ValueOnly(ctx)

Reply:
Changed as you suggested and also removed `<-ctx.Done()` inside, since it doesn't do anything